org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/snippeteditor/JavaSnippetEditor.java

```diff
@@ -636,8 +636,8 @@ protected void fireEvalStateChanged() {
 		Runnable r= new Runnable() {
 			@Override
 			public void run() {
-				Shell shell= getShell();
-				if (JavaSnippetEditor.this.fSnippetStateListeners != null && shell != null && !shell.isDisposed()) {
+				if (JavaSnippetEditor.this.fSnippetStateListeners != null) {
+					JavaSnippetEditor.this.dispose();
```

Comment:
Not sure how this matches PR description? 
Do you really want call `JavaSnippetEditor.this.dispose();` here?